index.js

```diff
@@ -1,8 +1,138 @@
 const battleship = () => {
   return 'The winner is...?'
+}  
+  var player1 = {
+  name : '',
+  shipCount: 4,
+  gameBoard: [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0]
+  ]
+};
+var player2 = {
+  name : '',
+  shipCount: 4,
+  gameBoard: [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0]
+  ]
+};
+var x;
+var y;
+function getRandomNumber() {
+  var valmin=0; 
+  var valmax=4;  
+  var num = Math.floor(Math.random() * (+valmax - +valmin)) + +valmin; 
+  return num;
 }
 
-const gameResult = battleship()
+function createShip(player, ships) {
+  let i = 0;
+  do {
+    const x = getRandomNumber();
+    const y = getRandomNumber();
+    if (player.gameBoard[x][y] !== 1) {
+      player.gameBoard[x][y] = 1;
+      i++;
+    }
+  } while (i < ships)
+}
+//Function to shot and count ships
+function disparar(player,x,y){
+  if (player.gameBoard[x][y] === 1){
+    console.log("estas disparando")
+    console.log(player)
+    //alert ("You hit your opponent has one less ship")
+    player.gameBoard[x][y] = 0
+    player.shipCount --
+  }else{
+    console.log ("lo siento no estas hundiendo barcos de tu oponente")
+    //alert("Sorry you did not hit your opponent")
+    console.log(player)
+  }
+  return player
+}
+// Create a function to give turns to the Players
+function turnos (currentplayer){//always starts with player1 by default
+var winner
+  do{
```

Comment:
This do-while loop has a lot of lines of code. It would be better splitting this code into several functions. That would make the code more readable.